openwisp_monitoring/device/models.py

```diff
@@ -98,11 +99,13 @@ def data(self):
         """
         if self.__data:
             return self.__data
-        q = (
-            "SELECT data FROM {0}.{1} WHERE pk = '{2}' "
-            "ORDER BY time DESC LIMIT 1".format(SHORT_RP, self.__key, self.pk)
-        )
-        points = list(query(q).get_points())
+        if TIMESERIES_DB['DATABASE'] == 'influxdb':
+            q = (
+                "SELECT data FROM {0}.{1} WHERE pk = '{2}' "
+                "ORDER BY time DESC LIMIT 1".format(SHORT_RP, self.__key, self.pk)
+            )
+        # TODO: Might be different for the other database
+        points = list(TimeseriesDB().query(q).get_points())
```

Comment:
the model doesn't have to know anything about influxdb or prometheus, it just have to call a method on the timeseries db backend to do what it needs to do, the differences for each timeseries DB are handled in its respective backend, which for the moment for us will be influxdb only.